testsuite/tests/publish/multiple-remotes/test.py

```diff
@@ -0,0 +1,407 @@
+"""
+Check publishing from a local repo with multiple remotes configured.
+"""
+
+
+import os
+import re
+import subprocess
+from typing import Optional
+
+from drivers.alr import run_alr, init_local_crate
+from drivers.asserts import assert_match
+from drivers.helpers import git_branch, git_head, init_git_repo
+
+
+TEST_ROOT_DIR = os.getcwd()
+
+COMMIT_CONFIRM_PROMPT = (
+    r".*The specified branch is not configured to track a remote\.(\r\n|\r|\n)"
+    r"Do you want to attempt to publish its head commit anyway\?"
+)
+COMMIT_NOT_PRESENT_MESSAGE = (
+    r".*The specified commit is not present on any configured remote\."
+)
+URL_CONFIRM_PROMPT_TEMPLATE = (
+    r".*The repository has multiple remotes configured, but the specified "
+    r"commit is only present on the remote 'remote{n}'\.(\r\n|\r|\n)"
+    r"The published manifest will therefore use the origin "
+    rf"'{re.escape(TEST_ROOT_DIR + os.path.sep)}remote{{n}}'\.(\r\n|\r|\n)"
+    r"Is this correct\?"
+)
+AMBIGUOUS_REMOTE_MESSAGE = (
+    r".*The specified commit is present on multiple remotes.*Please use "
+    r"'alr publish <remote-URL> <commit>' to resolve this ambiguity\."
+)
+UNSTAGED_CHANGES_MESSAGE = (
+    r".*You have unstaged changes\. Please commit or stash them\."
+)
+AHEAD_OF_REMOTE_MESSAGE = (
+    r".*Your branch is ahead of remote.*Please push local commits to the "
+    r"remote branch\."
+)
+BEHIND_REMOTE_CONFIRM_PROMPT = (
+    r".*There are commits on the remote branch which could be retrieved with "
+    r"'git pull'\.(\r\n|\r|\n)Are you sure you want to publish without them\?"
+)
+
+
+def run(*args, **kwargs):
+    subprocess.run(*args, **kwargs).check_returncode()
+
+def commit_file(commit_name: str, path: str, content: str):
+    """
+    Create a new file with the specified content and `git commit` it.
+
+    Also returns the commit's hash and attaches the tag `f"tag_{commit_name}"`
+    thereto.
+    """
+    with open(path, "x") as f:
+        f.write(content)
+    run(["git", "add", path])
+    run(["git", "commit", "-m", f"Commit {commit_name}"])
+    run(["git", "tag", f"tag_{commit_name}"])
+    return git_head()
```

Comment:
Should this be in helpers.py named as `git_commit_file`, for general availability?